### PR TITLE
feat: add table reader

### PR DESCRIPTION
**Pull request by farbodahm**
This PR:

- Add a table reader class
- Add related tests

### CODE REVIEW

File: src/sparkle/reader/table_reader.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        )
+
+    @property
+    def qualified_table_name(self) -> str:
+        """Gets the fully qualified table name.
+
+        Returns:
+            str: The fully qualified table name in the format "catalog_name.database_name.table_name".
+        """
+        return f"{self.catalog_name}.{self.database_name}.{self.table_name}"
```
Comment (shahinism) [inline src/sparkle/reader/table_reader.py:80]: Should we consider this format only works/is required for Iceberg tables?

File: src/sparkle/config/__init__.py [hunk 2]
```diff
@@ -13,7 +13,6 @@ class Config:
     version: str
     database_bucket: str
     kafka: KafkaConfig | None
-    input_database: TableConfig | None
-    output_database: TableConfig | None
-    iceberg_config: IcebergConfig | None
+    hive_table_input: TableConfig | None
+    iceberg_output_config: IcebergConfig | None
```
Comment (shahinism) [inline src/sparkle/config/__init__.py:17]:
```suggestion
    iceberg_output: IcebergConfig | None
```

Or maybe we should have it like:
```python
input: IcebergConfig | KafkaConfig | ...
output: IcebergConfig | KafkaConfig ...
```

File: src/sparkle/reader/table_reader.py (hunk 1 above)
Comment (farbodahm) [reply src/sparkle/reader/table_reader.py:80]: Look at test `test_read_table`. A normal table also can use this format.

Comment (farbodahm) [reply src/sparkle/reader/table_reader.py:80]: Have look at `test_read_table` test. Non-Iceberg table is also working with this format.

File: src/sparkle/config/__init__.py (hunk 2 above)
Comment (farbodahm) [reply src/sparkle/config/__init__.py]: Done

**Pull request merged**